libfwupdplugin/fu-fmap-firmware.c

```diff
@@ -0,0 +1,278 @@
+/*
+ * Copyright (C) 2020 Benson Leung <bleung@chromium.org>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include "fu-fmap-firmware.h"
+
+struct _FuFmapFirmware {
+	FuFirmwareClass		 parent_instance;
+};
+
+G_DEFINE_TYPE (FuFmapFirmware, fu_fmap_firmware, FU_TYPE_FIRMWARE)
+
+#define FMAP_SIGNATURE		"__FMAP__"
+#define FMAP_VER_MAJOR		1	/* this header's FMAP minor version */
+#define FMAP_VER_MINOR		1	/* this header's FMAP minor version */
+#define FMAP_STRLEN		32	/* maximum length for strings, */
+					/* including null-terminator */
+
+/* mapping of volatile and static regions in firmware binary */
+struct fmap_area {
+	guint32	offset;                /* offset relative to base */
+	guint32	size;                  /* size in bytes */
+	guint8	name[FMAP_STRLEN];     /* descriptive name */
+	guint16	flags;                 /* flags for this area */
+}  __attribute__((packed));
+
+struct fmap {
```

Comment:
in fwupd a `typedef struct` is more usual, also typically called namespaced something like FuFmap or something.

Reply:
Done.